Framework/src/runUploadRootObjects.cxx

```diff
@@ -0,0 +1,146 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    runUploadRootObjects.cxx
+/// \author  Piotr Konopka
+///
+/// \brief This is an executable which reads ROOT files with objects and puts them to QCDB.
+///
+/// This is an executable which reads QAResults.root generated by DPL analysis tasks and puts them to QCDB.
+/// It will ignore the directory structure and put all objects in under the task name specified as the argument.
+/// By default the current date and time will be used as the start of validity, and the object will be valid for 10 years.
+
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/CcdbDatabase.h"
+#include "QualityControl/MonitorObject.h"
+
+#include <functional>
+#include <string>
+#include <boost/program_options.hpp>
+#include <boost/exception/diagnostic_information.hpp>
+#include <TFile.h>
+#include <TKey.h>
+
+namespace bpo = boost::program_options;
+using namespace o2::quality_control::core;
+using namespace o2::quality_control::repository;
+
+int main(int argc, const char* argv[])
+{
+  size_t objectsUploaded = 0;
+
+  try {
+    bpo::options_description desc{ "Options" };
+    desc.add_options()                                                                                                              //
+      ("help,h", "Help screen")                                                                                                     //
+      ("input-file", bpo::value<std::string>()->default_value("./QAResults.root"), "Path to the ROOT file with objects to insert.") //
+      ("qcdb-url", bpo::value<std::string>()->default_value("ccdb-test.cern.ch:8080"), "URL to the QCDB.")                          //
+      ("task-name", bpo::value<std::string>(), "Name of the task which the objects belong to. Use / to make directories")           //
+      ("detector-code", bpo::value<std::string>()->default_value("TST"), "3-letter detector code. Put AOD for analysis tasks")      //
+      ("validity-start", bpo::value<uint64_t>()->default_value(0), "Start of objects validity in ms since epoch")                   //
+      ("validity-end", bpo::value<uint64_t>()->default_value(0), "End of objects validity in ms since epoch")                       //
```

Comment:
do you actually want default values here, in particular 0 ?

Reply:
For validity and filters: I don't think there is a good way to retrieve those values from analysis tasks, so I would not bother people too much about it before we can offer a solution.
For run number: indeed we could require an explicitly stated value, I will fix this. If it is multi-run, people should be aware of it and put 0 then.